xray/commands/audit/generic/jas/applicabilityscanner.go

```diff
@@ -0,0 +1,336 @@
+	return e.XrayResults
+}
+
+func GetExtendedScanResults(results []services.ScanResponse, dependencyTrees []*services.GraphNode,
+	serverDetails *config.ServerDetails) (*ExtendedScanResults, error) {
+	err := createAnalyzerManagerLogDir()
+	if err != nil {
+		return nil, err
+	}
+	applicabilityScanManager, err := NewApplicabilityScanManager(results, dependencyTrees, serverDetails)
+	if err != nil {
+		log.Info(applicabilityScanFailureMessage + err.Error())
+		return nil, err
+	}
+	shouldRun, err := applicabilityScanManager.shouldRun()
+	if err != nil {
+		log.Info(applicabilityScanFailureMessage + err.Error())
+		return nil, err
+	}
+	if !shouldRun {
+		log.Info("user not entitled for jas, didnt execute applicability scan")
+		return &ExtendedScanResults{XrayResults: results, ApplicabilityScannerResults: nil, EntitledForJas: false}, nil
+	}
+	entitledForJas, err := applicabilityScanManager.Run()
+	if !entitledForJas {
+		log.Info("got not entitled error from analyzer manager")
+		return &ExtendedScanResults{XrayResults: results, ApplicabilityScannerResults: nil, EntitledForJas: false}, nil
+	}
+	if err != nil {
+		log.Info(applicabilityScanFailureMessage + err.Error())
+		return nil, err
+	}
+	applicabilityScanResults := applicabilityScanManager.getApplicabilityScanResults()
+	extendedScanResults := ExtendedScanResults{XrayResults: results, ApplicabilityScannerResults: applicabilityScanResults, EntitledForJas: true}
+	return &extendedScanResults, nil
+}
+
+func (a *ApplicabilityScanManager) shouldRun() (bool, error) {
+	analyzerManagerExist, err := a.analyzerManager.DoesAnalyzerManagerExecutableExist()
+	if err != nil {
+		return false, err
+	}
+	return analyzerManagerExist && a.resultsIncludeEligibleTechnologies() && len(a.createCveList()) > 0, nil
+}
+
+func (a *ApplicabilityScanManager) resultsIncludeEligibleTechnologies() bool {
+	for _, vuln := range a.xrayVulnerabilities {
+		for _, technology := range technologiesEligibleForApplicabilityScan {
+			if vuln.Technology == technology.ToString() {
+				return true
+			}
+		}
+	}
+	for _, violation := range a.xrayViolations {
+		for _, technology := range technologiesEligibleForApplicabilityScan {
+			if violation.Technology == technology.ToString() {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+type ApplicabilityScanManager struct {
+	applicabilityScannerResults map[string]string
+	xrayVulnerabilities         []services.Vulnerability
+	xrayViolations              []services.Violation
+	configFileName              string
+	resultsFileName             string
+	analyzerManager             AnalyzerManager
+	serverDetails               *config.ServerDetails
+}
+
+func NewApplicabilityScanManager(xrayScanResults []services.ScanResponse, dependencyTrees []*services.GraphNode,
+	serverDetails *config.ServerDetails) (*ApplicabilityScanManager, error) {
+	directDependencies := getDirectDependenciesList(dependencyTrees)
+	tempDir, err := fileutils.CreateTempDir()
```

Comment:
The `NewApplicabilityScanManager` should create a `cleanup` function that deletes the temp dir. Otherwise, the temp dir will never be deleted.
See this function as an example for this - 
```
func (dc *DotnetCommand) prepareConfigFileIfNeeded() (cleanup func() error, err error)
```